datadog/fwprovider/resource_datadog_app_key_registration.go

```diff
@@ -0,0 +1,164 @@
+	response.TypeName = "app_key_registration"
+}
+
+func (r *appKeyRegistrationResource) Schema(_ context.Context, _ resource.SchemaRequest, response *resource.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Description: "Registers App Keys to be used for Action Connection, App Builder, and Workflow Automation. This registration is required to enable API and Terraform use in these products.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The Application Key ID to register.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // This replace means if the ID is changed the previous app key registration will be deleted and a new one is created.
+				},
+			},
+		},
+	}
+}
+
+func (r *appKeyRegistrationResource) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, frameworkPath.Root("id"), request, response)
+}
+
+func (r *appKeyRegistrationResource) Create(ctx context.Context, request resource.CreateRequest, response *resource.CreateResponse) {
+	var plan appKeyRegistrationResourceModel
+	diags := request.Plan.Get(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	id, err := uuid.Parse(plan.ID.ValueString())
+	if err != nil {
+		response.Diagnostics.AddError("error parsing ID as UUID", err.Error())
+		return
+	}
+
+	_, httpResp, err := r.Api.RegisterAppKey(r.Auth, id.String())
+	if err != nil {
+		if httpResp != nil {
+			body, err := io.ReadAll(httpResp.Body)
+			if err != nil {
+				response.Diagnostics.AddError("Error reading error response when registering app key", err.Error())
+				return
+			}
+			response.Diagnostics.AddError("Error registering app key", string(body))
+		} else {
+			response.Diagnostics.AddError("Error registering app key", err.Error())
+		}
+		return
+	}
+
+	diags = response.State.Set(ctx, &plan)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *appKeyRegistrationResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var state appKeyRegistrationResourceModel
+	diags := request.State.Get(ctx, &state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	appKeyRegistration, httpResp, err := r.Api.GetAppKeyRegistration(r.Auth, state.ID.ValueString())
+	if err != nil {
+		if httpResp != nil {
+			body, err := io.ReadAll(httpResp.Body)
+			if err != nil {
+				response.Diagnostics.AddError("Error reading error response when getting app key registration", err.Error())
+				return
+			}
+
+			// If the app key registration is not found, we log a warning and remove the resource from state. This may be due to changes in the UI or attempting to import an app key that is not registered.
```

Comment:
I think having a "read" remove the resource from state is unintuitive. I think we should keep it as a 404 and it'll be on the customer to remove any non-existent registrations from their TF config.